libcudacxx/test/libcudacxx/cuda/iterators/transform_iterator/subscript.pass.cpp

```diff
@@ -0,0 +1,96 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+// transform_iterator::operator[]
+
+#include <cuda/iterator>
+#include <cuda/std/cassert>
+
+#include "test_iterators.h"
+#include "test_macros.h"
+#include "types.h"
+
+template <class Iter>
+_CCCL_CONCEPT can_subscript = _CCCL_REQUIRES_EXPR((Iter), Iter i)((i[0]));
+
+template <class Iter>
+__host__ __device__ constexpr void test()
+{
+  if constexpr (cuda::std::random_access_iterator<Iter>)
+  {
+    int buffer[8] = {0, 1, 2, 3, 4, 5, 6, 7};
+
+    {
+      cuda::transform_iterator iter{Iter{buffer}, PlusOne{}};
+      assert(iter[4] == 5);
+      assert(buffer[4] == 4);
```

Comment:
Suggestion: can we add a test that *writes* through a transform iterator? E.g., using a projection.
```c++
cuda::transform_iterator iter{Iter{buffer}, &Class::member};
iter[0] = 42; // write
```

Reply:
Was added below.